### ISSUE TITLE
Use a digital input to call the /startmash and/or /proceed endpoints

### ISSUE THREAD

**Issue by astephon88** [fdewasmes/brewblox-brewfather-service#2]
It would be quite handy to tie a Spark digital input into the automation, allowing for a button press to start the mash automation, and continue the automation from a pause

**Comment by fdewasmes** [comment]
in fact, there is already an API call ready to resume. When you are talking about a digital input, do you mean a physical IO?

**Comment by astephon88** [comment]
Correct. I have a push button on my control panel that I could connect to a spark 4 GPIO pin configured as an input, and would like to be able to use that to start/resume the automation

**Comment by fdewasmes** [comment]
I know Elco was planning to have this kind of small automation directly in the spark itself together with a piezo alarm. I asked him a few days back but didn't receive an answer and I'm not sure this is high in his priority list.
I think this is the kind of automation that could be easily and properly handled using the embedded Node-red as I guess digital input raises a message on MQTT. Am I right? If you can provide me with a payload sample I could try to setup a node red schema for this.

**Comment by astephon88** [comment]
I haven't seen any documentation for the digital inputs, and I haven't received my spark 4 yet, so I don't have an answer there. I was envisioning this "resume via digital input" feature as being an optional part of the brewfather automation itself (i.e., you specify what service/pin to use as the resume with a parameter in the compose file, and if specified, the automation will monitor the input for events and act accordingly). But if you don't think that's appropriate, I can look at writing a separate service, or using node-red. Are there any examples of using node-red with brewblox? I've only seen it mentioned in the design decision document about deprecating the automation service.

**Comment by astephon88** [comment]
So, this might be a moot point for now. I just got my Spark 4, and I don't see any way to configure a pin as an input. Maybe that hasn't actually been implemented yet?

**Comment by fdewasmes** [comment]
Normally you should be able to configure it this way but this may not be available on the UI yet. Have you tried to contact them on Discord?

**Comment by fdewasmes** [comment]
I can surely make an update to have automation resume using this input pin. I think this is useful for anyone.

**Comment by astephon88** [comment]
Just asked on discord. Elco said that's what he's currently working on, so it's not available yet. Sound like he's trying to wrap his head around use cases, so I pointed him to this issue.
